File: day_eight/fibonacci.cpp
#include <iostream>
//Not really meant to handle big integers

int main() {
    int firstNum = 0;
    int secondNum = 1;
    int userChoice;

    std::cout << "How many elements of the Fibonacci sequence do you want me to print out?\n";
    std::cin >> userChoice;

    if (userChoice == 0) {
        std::cout << "Thank you for wasting my time!" << '\n';
    } 
    else if (userChoice == 1) {
        std::cout << firstNum << '\n';
    } 
    else if (userChoice == 2) {
        std::cout << firstNum << '\n' << secondNum << '\n';
    } 
    else {
        int count = 2;
        int temp;
        std::cout << firstNum << '\n' << secondNum << '\n';
        while (count < userChoice) {
            std::cout << firstNum + secondNum << '\n';
            temp = secondNum;
            secondNum = firstNum + secondNum;
            firstNum = temp;
            count++;
        }
    }
    return 0;
}

File: day_eight/fibonacci.py
#!/usr/bin/python3

import sys

def fibonacci(num):
    first_num = 0
    second_num = 1
    fibonacci = [first_num, second_num]
    count = 1

    if num == 1:
        return fibonacci[0]
    elif num == 2:
        return fibonacci
    else:
        while count <= num:
            fibonacci.append(second_num + first_num)
            temp = second_num
            second_num = first_num + second_num
            first_num = temp
            count+=1
    
        return fibonacci[:num]

def main():
    num = int(sys.argv[1])
    print(fibonacci(num))

if __name__=="__main__":
    main()